engine/apps/api/views/user_notification_policy.py

```diff
@@ -54,25 +73,27 @@ class UserNotificationPolicyView(UpdateSerializerMixin, OrderedModelViewSet):
             "move_to_position",
         ],
     }
+    queryset = UserNotificationPolicy.objects.none()  # needed for drf-spectacular introspection
 
     model = UserNotificationPolicy
     serializer_class = UserNotificationPolicySerializer
     update_serializer_class = UserNotificationPolicyUpdateSerializer
 
+    filter_backends = (filters.DjangoFilterBackend,)
+    filterset_class = UserNotificationPolicyFilter
+
     def get_queryset(self):
-        important = self.request.query_params.get("important", None) == "true"
-        try:
-            user_id = self.request.query_params.get("user", None)
-        except ValueError:
-            raise BadRequest(detail="Invalid user param")
-        if user_id is None or user_id == self.request.user.public_primary_key:
-            target_user = self.request.user
-        else:
-            try:
-                target_user = User.objects.get(public_primary_key=user_id)
-            except User.DoesNotExist:
-                raise BadRequest(detail="User does not exist")
-        queryset = UserNotificationPolicy.objects.filter(user=target_user, important=important)
+        # if there are no query params, set default value
+        lookup_kwargs = {}
+        important = self.request.query_params.get("important", None)
+        user_id = self.request.query_params.get("user", None)
+        if important is None:
+            lookup_kwargs.update({"important": False})
+        if user_id is None:
+            lookup_kwargs.update({"user": self.request.user})
+        queryset = UserNotificationPolicy.objects.filter(
+            **lookup_kwargs, user__organization=self.request.auth.organization
+        )
```

Comment:
doesn't this apply filters 2 times? first time in `get_queryset` and second time in `filter_queryset` under the hood ([ListModelMixin](https://github.com/encode/django-rest-framework/blob/9921c7554f3d8ce435e45bafd27f4b241beb66f1/rest_framework/mixins.py#L38))

Reply:
In `get_queryset` we use the default value to filter queryset, when the parameter is not present. In `filter_queryset` filter for not presented params will be skipped.